### ISSUE TITLE
Create a CONTRIBUTING document

### ISSUE THREAD

**Issue by mriccid** [react-csv/react-csv#112]
A contributing guide would be helpful in ensuring that new issues and pull requests are well considered, necessary and are within scope. Some issues that are being created could be avoided by having a clear document that lays out things like 
- how to create a clear bug report,
- some guidelines for feature requests,
- tips on how to avoid common pitfalls (e.g. avoiding problems caused by malformed data!),
- requirements for pull requests (e.g. unit tests, coverage, documentation)

**Comment by mccabemj** [comment]
@mriccid agreed. Both in the documentation and [here](https://github.com/react-csv/react-csv/community)

I definitely like the idea of including tips to avoid common mistakes within the documentation. Some kind of troubleshooting guide, or at least expand upon the documentation to make it more detailed for new users.

I'm happy to work on the community section here. Perhaps you'd be best placed for troubleshooting documentation?
